client/js/stores/snippet-store.js

```diff
@@ -38,6 +38,40 @@ class SnippetStore {
     this.preventDefault();
   }
 
+  countSnippetAverage(data) {
+    var snippetId = data.snippetId;
+    var userRating = data.userRate;
+    var avg = data.avarage;
+    var snippetsAverage = this.state.snippetsAvg;
+
+    snippetsAverage[snippetId] = avg;
```

Comment:
why do we need array for storing average if each snippet already has it as attribute? Also I would consider moving recalculating logic to action